undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -408,14 +417,31 @@ public Duration getDefaultSessionTimeout() {
             };
             CapabilityServiceConfigurator factoryConfigurator = provider.getSessionManagerFactoryServiceConfigurator(factoryServiceName, configuration);
             CapabilityServiceConfigurator codecConfigurator = provider.getSessionIdentifierCodecServiceConfigurator(codecServiceName, configuration);
+            CapabilityServiceConfigurator affinityConfigurator = provider.getAffinityLocatorServiceConfigurator(affinityServiceName, configuration);
 
             smfSupplier = udisBuilder.requires(factoryConfigurator.getServiceName());
             sicSupplier = udisBuilder.requires(codecConfigurator.getServiceName());
+            affinityLocatorSupplier = udisBuilder.requires(affinityConfigurator.getServiceName());
+
+            CookieConfig affinityCookieConfig = (servletContainer != null) ? servletContainer.getAffinityCookieConfig() : null;
```

Comment:
FYI: https://github.com/wildfly/wildfly/pull/16539

Reply:
I have dropped my local changes fixing this and incorporated https://github.com/wildfly/wildfly/pull/16539 instead and rebased.